components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/factory/ScopeServiceFactory.java

```diff
@@ -0,0 +1,58 @@
+package org.wso2.carbon.identity.oauth.endpoint.factory;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.config.AbstractFactoryBean;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.oauth.scopeservice.ScopeService;
+import org.wso2.carbon.identity.oauth2.OAuth2ScopeService;
+
+import java.util.List;
+
+/**
+ * Factory bean for ScopeService.
+ */
+public class ScopeServiceFactory extends AbstractFactoryBean<ScopeService> {
+
+    private ScopeService scopeService;
+
+    private static final Log log = LogFactory.getLog(ScopeServiceFactory.class);
+
+    @Override
+    public Class<ScopeService> getObjectType() {
+
+        return ScopeService.class;
+    }
+
+    @Override
+    protected ScopeService createInstance() throws Exception {
+
+        if (this.scopeService == null) {
+
+            // Get the OSGi service registered for ScopeService interface.
+            List<Object> scopeServices = PrivilegedCarbonContext
+                    .getThreadLocalCarbonContext().getOSGiServices(ScopeService.class, null);
+
+            if (scopeServices == null || scopeServices.isEmpty()) {
+                log.debug("No ScopeService is available. Returning OAuth2ScopeService instead.");
+                return new OAuth2ScopeService();
```

Comment:
Will this set the scopeService instance variable?
or Will this return a new instance everytime?